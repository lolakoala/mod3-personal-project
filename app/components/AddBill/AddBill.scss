.add-bill {
  margin-left: 20px;
}

.add-bill input,
.add-bill button,
.add-bill textarea {
  font-family: 'BenchNine', sans-serif;
  border-radius: 25px;
  margin-top: 20px;
  height: 25px;
  font-size: 20px;
  width: 120px;
  margin-left: 10px;
  margin-right: 10px;
}

.add-bill input,
.add-bill textarea {
  padding-left: 10px;
  border: 1px solid #D83D59;
  color: #D83D59;
  border-radius: 5px;
}

.add-bill input::placeholder,
.add-bill textarea::placeholder {
  color: #D83D59;
}

.add-bill button {
  border: none;
  background-color: #D83D59;
  color: #F4CCE2;
}

.add-bill button:hover,
.add-bill .active {
  cursor: pointer;
  border: 2px dashed #D83D59;
  background-color: #F4CCE2;
  color: #D83D59;
}

.add-bill .split {
  font-size: 10px;
}

.add-bill,
.add-bill-inputs,
.split-buttons,
.form-buttons {
  display: flex;
  flex-direction: column;
}

.inputs-and-splits,
.details-and-buttons {
  display: flex;
}

.add-bill textarea {
  height: 70px;
}

.add-bill .split-info {
  margin-left: 20px;
  font-family: 'BenchNine', sans-serif;
  font-size: 20px;

}
