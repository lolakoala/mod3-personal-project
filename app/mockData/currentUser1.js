const currentUser1 = {
  id : "RVde1m9BhkQBLZlTJNsFlQZXqP43",
  name: "Amanda Brenner",
  house: "Dog9"
};

export default currentUser1;
