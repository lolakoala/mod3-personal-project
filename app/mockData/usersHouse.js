import currentUser1 from './currentUser1.js';
import currentUser2 from './currentUser2.js';

const bill1 = {
  allUsersTotals: [
    {
      id: currentUser1.id,
      paid: false,
      total: 50
    },
    {
      id: currentUser2.id,
      paid: false,
      total: 50
    }
  ],
  datePosted: "11/01/2017",
  details: "We used a lot of water.",
  duedate: "11/9/17",
  id: "b3019762705779",
  parsedDuedate: 1510210800000,
  postedBy: currentUser1,
  title: "Water Bill",
  total: "100"
};

const bulletin1 = {
  datePosted: "11/01/2017",
  details: "The dogs got out through a hole in the fence yesterday.",
  hasRead: [currentUser1.id],
  id: 1509567098747,
  postedBy: currentUser1,
  title: "Don't leave dogs out unattended."
};

const chore1 = {
  assignedTo: currentUser1.id,
  datePosted: "10/31/2017",
  details: "Empty the dishwasher.",
  done: false,
  id: "c1509485241076",
  postedBy: currentUser1,
  title: "Dishes",
  urgency: "low"
};

const usersHouse = {
  houseKey: "-KxoIBWy5UYRscFYdWty",
  bills: [bill1],
  bulletins: [bulletin1],
  chores: [chore1],
  houseCode: "woof9",
  houseName: "Dog9",
  users: [currentUser1, currentUser2]
};

export default usersHouse;
