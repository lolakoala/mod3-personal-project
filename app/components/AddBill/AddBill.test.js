import React from "react";
import AddBill from './AddBill.js';
import { shallow } from "enzyme";
import currentUser1 from '../../mockData/currentUser1.js';
import usersHouse from '../../mockData/usersHouse.js';

describe('AddBill component', () => {
  const mockFn = jest.fn();
  const wrapper = shallow(<AddBill usersHouse={usersHouse}
    currentUser={currentUser1}
    addBill={mockFn}
    getTodaysDate={mockFn}
    updateItem={mockFn}/>);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should have a default state', () => {
    const defaultState = {
      item: '',
      split: '',
      equalSelect: [],
      title: '',
      duedate: '',
      total: '',
      allUsersTotals: [],
      details: '',
      buttonDisabled: true,
      error: '',
      percentageLeft: 100,
      dollarLeft: ''
    };

    expect(wrapper.state()).toEqual(defaultState);
  });

  it('should have 6 buttons', () => {
    const buttons = wrapper.find('button');

    expect(buttons.length).toEqual(6);
  });

  it('should start with 3 inputs', () => {
    const inputs = wrapper.find('input');

    expect(inputs.length).toEqual(3);
  });

  it('should change state', () => {
    const titleInput = wrapper.find('.title-input');
    const duedateInput = wrapper.find('.duedate-input');
    const totalInput = wrapper.find('.total-input');
    const detailsInput = wrapper.find('textarea');

    titleInput.simulate('change', { target: { value: 'Gas Bill' } });
    duedateInput.simulate('change', { target: { value: '11/9/17' } });
    totalInput.simulate('change', { target: { value: '150' } });
    detailsInput.simulate('change', { target: { value: 'We turned on the heat this month.' } });
    expect(wrapper.state().title).toEqual('Gas Bill');
    expect(wrapper.state().duedate).toEqual('11/9/17');
    expect(wrapper.state().total).toEqual('150');
    expect(wrapper.state().details).toEqual('We turned on the heat this month.');
  });

  it('should calculate an equal split', () => {
    const equalAllButton = wrapper.find('.equal-all-button');
    const { users } = usersHouse;
    const userTotal = wrapper.state().total / users.length;
    const newState = users.map(user => ({
      id: user.id,
      total: userTotal,
      paid: false
    }));

    equalAllButton.simulate('click');
    expect(wrapper.state().split).toEqual('equalAll');
    expect(wrapper.state().allUsersTotals).toEqual(newState);
    expect(wrapper.state().buttonDisabled).toEqual(false);
  });

  it('should calculate a custom equal split', () => {
    const equalSelectButton = wrapper.find('.equal-select-button');

    equalSelectButton.simulate('click');
    const inputs = wrapper.find('input');

    expect(inputs.length).toEqual(3 + usersHouse.users.length);
    expect(wrapper.state().split).toEqual('equal');
  });

  it('should calculate custom split by percentage', () => {
    const customPercentageButton = wrapper.find('.custom-percentage-button');

    customPercentageButton.simulate('click');
    const inputs = wrapper.find('input');

    expect(inputs.length).toEqual(3 + usersHouse.users.length);
    expect(wrapper.state().split).toEqual('customP');
  });

  it('should calculate custom split by dollar', () => {
    const customDollarButton = wrapper.find('.custom-dollar-button');

    customDollarButton.simulate('click');
    const inputs = wrapper.find('input');


    expect(inputs.length).toEqual(3 + usersHouse.users.length);
    expect(wrapper.state().split).toEqual('customD');
  });

  it('should fire functions', () => {
    const equalAllButton = wrapper.find('.equal-all-button');
    const billSubmitButton = wrapper.find('.bill-submit');

    equalAllButton.simulate('click');
    billSubmitButton.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(3);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
