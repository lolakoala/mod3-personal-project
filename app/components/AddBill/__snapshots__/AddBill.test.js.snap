// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBill component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddBill
    addBill={[Function]}
    currentUser={
      Object {
        "house": "Dog9",
        "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
        "name": "Amanda Brenner",
      }
    }
    getTodaysDate={[Function]}
    updateItem={[Function]}
    usersHouse={
      Object {
        "bills": Array [
          Object {
            "allUsersTotals": Array [
              Object {
                "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                "paid": false,
                "total": 50,
              },
              Object {
                "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
                "paid": false,
                "total": 50,
              },
            ],
            "datePosted": "11/01/2017",
            "details": "We used a lot of water.",
            "duedate": "11/9/17",
            "id": "b3019762705779",
            "parsedDuedate": 1510210800000,
            "postedBy": Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            },
            "title": "Water Bill",
            "total": "100",
          },
        ],
        "bulletins": Array [
          Object {
            "datePosted": "11/01/2017",
            "details": "The dogs got out through a hole in the fence yesterday.",
            "hasRead": Array [
              "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            ],
            "id": 1509567098747,
            "postedBy": Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            },
            "title": "Don't leave dogs out unattended.",
          },
        ],
        "chores": Array [
          Object {
            "assignedTo": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            "datePosted": "10/31/2017",
            "details": "Empty the dishwasher.",
            "done": false,
            "id": "c1509485241076",
            "postedBy": Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            },
            "title": "Dishes",
            "urgency": "low",
          },
        ],
        "houseCode": "woof9",
        "houseKey": "-KxoIBWy5UYRscFYdWty",
        "houseName": "Dog9",
        "users": Array [
          Object {
            "house": "Dog9",
            "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            "name": "Amanda Brenner",
          },
          Object {
            "house": "Dog9",
            "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
            "name": "Lola Koala",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="inputs-and-splits"
        >
          <div
            className="add-bill-inputs"
          >
            <input
              className="bill-input title-input"
              onChange={[Function]}
              placeholder="Title"
              type="text"
            />
            <input
              className="bill-input duedate-input"
              onChange={[Function]}
              placeholder="Due Date"
              type="text"
            />
            <input
              className="bill-input total-input"
              onChange={[Function]}
              placeholder="Total"
              type="text"
            />
          </div>
          <div
            className="split-buttons"
          >
            <button
              className="split equal-all-button active"
              onClick={[Function]}
            >
              Equal Split with All House Members
            </button>
            <button
              className="split equal-select-button "
              onClick={[Function]}
            >
              Equal Split with Select House Members
            </button>
            <button
              className="split custom-percentage-button "
              onClick={[Function]}
            >
              Custom Split by Percentage
            </button>
            <button
              className="split custom-dollar-button "
              onClick={[Function]}
            >
              Custom Split by Dollar Amount
            </button>
          </div>
        </div>,
        <div
          className="split-info"
        >
          <p />
          <div>
            <p>
              Amanda Brenner
            </p>
            <p>
              75
            </p>
          </div>
          <div>
            <p>
              Lola Koala
            </p>
            <p>
              75
            </p>
          </div>
        </div>,
        <div
          className="details-and-buttons"
        >
          <textarea
            onChange={[Function]}
            placeholder="Details"
            type="text"
          />
          <div
            className="form-buttons"
          >
            <button
              className="bill-submit"
              disabled={false}
              onClick={[Function]}
            >
              Submit
            </button>
            <button
              onClick={[Function]}
            >
              Reset
            </button>
          </div>
        </div>,
      ],
      "className": "add-bill",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="add-bill-inputs"
            >
              <input
                className="bill-input title-input"
                onChange={[Function]}
                placeholder="Title"
                type="text"
              />
              <input
                className="bill-input duedate-input"
                onChange={[Function]}
                placeholder="Due Date"
                type="text"
              />
              <input
                className="bill-input total-input"
                onChange={[Function]}
                placeholder="Total"
                type="text"
              />
            </div>,
            <div
              className="split-buttons"
            >
              <button
                className="split equal-all-button active"
                onClick={[Function]}
              >
                Equal Split with All House Members
              </button>
              <button
                className="split equal-select-button "
                onClick={[Function]}
              >
                Equal Split with Select House Members
              </button>
              <button
                className="split custom-percentage-button "
                onClick={[Function]}
              >
                Custom Split by Percentage
              </button>
              <button
                className="split custom-dollar-button "
                onClick={[Function]}
              >
                Custom Split by Dollar Amount
              </button>
            </div>,
          ],
          "className": "inputs-and-splits",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="bill-input title-input"
                  onChange={[Function]}
                  placeholder="Title"
                  type="text"
                />,
                <input
                  className="bill-input duedate-input"
                  onChange={[Function]}
                  placeholder="Due Date"
                  type="text"
                />,
                <input
                  className="bill-input total-input"
                  onChange={[Function]}
                  placeholder="Total"
                  type="text"
                />,
              ],
              "className": "add-bill-inputs",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "bill-input title-input",
                  "onChange": [Function],
                  "placeholder": "Title",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "bill-input duedate-input",
                  "onChange": [Function],
                  "placeholder": "Due Date",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "bill-input total-input",
                  "onChange": [Function],
                  "placeholder": "Total",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="split equal-all-button active"
                  onClick={[Function]}
                >
                  Equal Split with All House Members
                </button>,
                <button
                  className="split equal-select-button "
                  onClick={[Function]}
                >
                  Equal Split with Select House Members
                </button>,
                <button
                  className="split custom-percentage-button "
                  onClick={[Function]}
                >
                  Custom Split by Percentage
                </button>,
                <button
                  className="split custom-dollar-button "
                  onClick={[Function]}
                >
                  Custom Split by Dollar Amount
                </button>,
              ],
              "className": "split-buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Equal Split with All House Members",
                  "className": "split equal-all-button active",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Equal Split with All House Members",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Equal Split with Select House Members",
                  "className": "split equal-select-button ",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Equal Split with Select House Members",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Custom Split by Percentage",
                  "className": "split custom-percentage-button ",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Custom Split by Percentage",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Custom Split by Dollar Amount",
                  "className": "split custom-dollar-button ",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Custom Split by Dollar Amount",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p />,
            Array [
              <div>
                <p>
                  Amanda Brenner
                </p>
                <p>
                  75
                </p>
              </div>,
              <div>
                <p>
                  Lola Koala
                </p>
                <p>
                  75
                </p>
              </div>,
            ],
            null,
            null,
            null,
            null,
          ],
          "className": "split-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": null,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Amanda Brenner
                </p>,
                <p>
                  75
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Amanda Brenner",
                },
                "ref": null,
                "rendered": "Amanda Brenner",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": 75,
                },
                "ref": null,
                "rendered": 75,
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Lola Koala
                </p>,
                <p>
                  75
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Lola Koala",
                },
                "ref": null,
                "rendered": "Lola Koala",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": 75,
                },
                "ref": null,
                "rendered": 75,
                "type": "p",
              },
            ],
            "type": "div",
          },
          null,
          null,
          null,
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <textarea
              onChange={[Function]}
              placeholder="Details"
              type="text"
            />,
            <div
              className="form-buttons"
            >
              <button
                className="bill-submit"
                disabled={false}
                onClick={[Function]}
              >
                Submit
              </button>
              <button
                onClick={[Function]}
              >
                Reset
              </button>
            </div>,
          ],
          "className": "details-and-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Details",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="bill-submit"
                  disabled={false}
                  onClick={[Function]}
                >
                  Submit
                </button>,
                <button
                  onClick={[Function]}
                >
                  Reset
                </button>,
              ],
              "className": "form-buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit",
                  "className": "bill-submit",
                  "disabled": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Submit",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Reset",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Reset",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="inputs-and-splits"
          >
            <div
              className="add-bill-inputs"
            >
              <input
                className="bill-input title-input"
                onChange={[Function]}
                placeholder="Title"
                type="text"
              />
              <input
                className="bill-input duedate-input"
                onChange={[Function]}
                placeholder="Due Date"
                type="text"
              />
              <input
                className="bill-input total-input"
                onChange={[Function]}
                placeholder="Total"
                type="text"
              />
            </div>
            <div
              className="split-buttons"
            >
              <button
                className="split equal-all-button active"
                onClick={[Function]}
              >
                Equal Split with All House Members
              </button>
              <button
                className="split equal-select-button "
                onClick={[Function]}
              >
                Equal Split with Select House Members
              </button>
              <button
                className="split custom-percentage-button "
                onClick={[Function]}
              >
                Custom Split by Percentage
              </button>
              <button
                className="split custom-dollar-button "
                onClick={[Function]}
              >
                Custom Split by Dollar Amount
              </button>
            </div>
          </div>,
          <div
            className="split-info"
          >
            <p />
            <div>
              <p>
                Amanda Brenner
              </p>
              <p>
                75
              </p>
            </div>
            <div>
              <p>
                Lola Koala
              </p>
              <p>
                75
              </p>
            </div>
          </div>,
          <div
            className="details-and-buttons"
          >
            <textarea
              onChange={[Function]}
              placeholder="Details"
              type="text"
            />
            <div
              className="form-buttons"
            >
              <button
                className="bill-submit"
                disabled={false}
                onClick={[Function]}
              >
                Submit
              </button>
              <button
                onClick={[Function]}
              >
                Reset
              </button>
            </div>
          </div>,
        ],
        "className": "add-bill",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="add-bill-inputs"
              >
                <input
                  className="bill-input title-input"
                  onChange={[Function]}
                  placeholder="Title"
                  type="text"
                />
                <input
                  className="bill-input duedate-input"
                  onChange={[Function]}
                  placeholder="Due Date"
                  type="text"
                />
                <input
                  className="bill-input total-input"
                  onChange={[Function]}
                  placeholder="Total"
                  type="text"
                />
              </div>,
              <div
                className="split-buttons"
              >
                <button
                  className="split equal-all-button active"
                  onClick={[Function]}
                >
                  Equal Split with All House Members
                </button>
                <button
                  className="split equal-select-button "
                  onClick={[Function]}
                >
                  Equal Split with Select House Members
                </button>
                <button
                  className="split custom-percentage-button "
                  onClick={[Function]}
                >
                  Custom Split by Percentage
                </button>
                <button
                  className="split custom-dollar-button "
                  onClick={[Function]}
                >
                  Custom Split by Dollar Amount
                </button>
              </div>,
            ],
            "className": "inputs-and-splits",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="bill-input title-input"
                    onChange={[Function]}
                    placeholder="Title"
                    type="text"
                  />,
                  <input
                    className="bill-input duedate-input"
                    onChange={[Function]}
                    placeholder="Due Date"
                    type="text"
                  />,
                  <input
                    className="bill-input total-input"
                    onChange={[Function]}
                    placeholder="Total"
                    type="text"
                  />,
                ],
                "className": "add-bill-inputs",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "className": "bill-input title-input",
                    "onChange": [Function],
                    "placeholder": "Title",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "className": "bill-input duedate-input",
                    "onChange": [Function],
                    "placeholder": "Due Date",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "className": "bill-input total-input",
                    "onChange": [Function],
                    "placeholder": "Total",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="split equal-all-button active"
                    onClick={[Function]}
                  >
                    Equal Split with All House Members
                  </button>,
                  <button
                    className="split equal-select-button "
                    onClick={[Function]}
                  >
                    Equal Split with Select House Members
                  </button>,
                  <button
                    className="split custom-percentage-button "
                    onClick={[Function]}
                  >
                    Custom Split by Percentage
                  </button>,
                  <button
                    className="split custom-dollar-button "
                    onClick={[Function]}
                  >
                    Custom Split by Dollar Amount
                  </button>,
                ],
                "className": "split-buttons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Equal Split with All House Members",
                    "className": "split equal-all-button active",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Equal Split with All House Members",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Equal Split with Select House Members",
                    "className": "split equal-select-button ",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Equal Split with Select House Members",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Custom Split by Percentage",
                    "className": "split custom-percentage-button ",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Custom Split by Percentage",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Custom Split by Dollar Amount",
                    "className": "split custom-dollar-button ",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Custom Split by Dollar Amount",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p />,
              Array [
                <div>
                  <p>
                    Amanda Brenner
                  </p>
                  <p>
                    75
                  </p>
                </div>,
                <div>
                  <p>
                    Lola Koala
                  </p>
                  <p>
                    75
                  </p>
                </div>,
              ],
              null,
              null,
              null,
              null,
            ],
            "className": "split-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": null,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Amanda Brenner
                  </p>,
                  <p>
                    75
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Amanda Brenner",
                  },
                  "ref": null,
                  "rendered": "Amanda Brenner",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": 75,
                  },
                  "ref": null,
                  "rendered": 75,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Lola Koala
                  </p>,
                  <p>
                    75
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Lola Koala",
                  },
                  "ref": null,
                  "rendered": "Lola Koala",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": 75,
                  },
                  "ref": null,
                  "rendered": 75,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            null,
            null,
            null,
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <textarea
                onChange={[Function]}
                placeholder="Details"
                type="text"
              />,
              <div
                className="form-buttons"
              >
                <button
                  className="bill-submit"
                  disabled={false}
                  onClick={[Function]}
                >
                  Submit
                </button>
                <button
                  onClick={[Function]}
                >
                  Reset
                </button>
              </div>,
            ],
            "className": "details-and-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Details",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="bill-submit"
                    disabled={false}
                    onClick={[Function]}
                  >
                    Submit
                  </button>,
                  <button
                    onClick={[Function]}
                  >
                    Reset
                  </button>,
                ],
                "className": "form-buttons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Submit",
                    "className": "bill-submit",
                    "disabled": false,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Submit",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Reset",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Reset",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
